File: java/Program/PerabotanRumahTangga.java
// =============================
// Class PerabotanRumahTangga
// Level 2 (Child dari ProdukElektronik)
// =============================

public class PerabotanRumahTangga extends ProdukElektronik {
    // =============================
    // Atribut tambahan khusus perabotan rumah tangga
    // =============================
    private String kategori;  // kategori produk (misalnya Interior, Outdoor)
    private String material;  // bahan dasar produk (Kayu, Besi, Plastik, dll)
    private float berat;      // berat produk dalam kg

    // =============================
    // Constructor kosong (default)
    // =============================
    public PerabotanRumahTangga() {}

    // =============================
    // Constructor dengan parameter
    // Memanggil parent constructor (super) untuk atribut dasar
    // =============================
    public PerabotanRumahTangga(String namaProduk, String merek, int harga,
                                String kategori, String material, float berat) {
        super(namaProduk, merek, harga); // panggil constructor parent
        this.kategori = kategori;
        this.material = material;
        this.berat = berat;
    }

    // =============================
    // Getter dan Setter
    // =============================
    public String getKategori() { return kategori; }
    public String getMaterial() { return material; }
    public float getBerat() { return berat; }

    public void setKategori(String kategori) { this.kategori = kategori; }
    public void setMaterial(String material) { this.material = material; }
    public void setBerat(float berat) { this.berat = berat; }
}

File: python/Program/PerabotanRumahTangga.py
from ProdukElektronik import ProdukElektronik

# =============================
# Class PerabotanRumahTangga
# Level 2 (Child dari ProdukElektronik)
# =============================
class PerabotanRumahTangga(ProdukElektronik):
    def __init__(self, nama_produk="", merek="", harga=0,
                 kategori="", material="", berat=0.0):
        """
        Inisialisasi atribut perabotan rumah tangga
        Memanggil constructor parent
        """
        super().__init__(nama_produk, merek, harga)  # panggil constructor parent
        self.kategori = kategori
        self.material = material
        self.berat = berat

    # Getter dan Setter
    def get_kategori(self):
        return self.kategori

    def set_kategori(self, kategori):
        self.kategori = kategori

    def get_material(self):
        return self.material

    def set_material(self, material):
        self.material = material

    def get_berat(self):
        return self.berat

    def set_berat(self, berat):
        self.berat = berat